tests/utils/secrets.go

```diff
@@ -48,3 +48,13 @@ func CreateSecretFromDefinition(c *kubernetes.Clientset, secret *v1.Secret) (*v1
 	}
 	return secret, nil
 }
+
+func DeleteSecret(clientSet *kubernetes.Clientset, namespace string, secret v1.Secret) error {
+	return wait.PollImmediate(SecretPollInterval, SecretPollPeriod, func() (bool, error) {
+		err := clientSet.CoreV1().Secrets(namespace).Delete(secret.GetName(), nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
```

Comment:
The return of a non-nil err stops `PollImmediate`

Reply:
so if we do not want to "stop polling", should we ignore the err and return nil instead?

Reply:
Yes. Polling stops 1st if err != nil and 2nd if done is true and 3rd if the overall time expires.

Reply:
Changed it a bit. Please let me know what do you think...

Reply:
Also added a call to this method in transport_test.go